File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    gender: string | undefined;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 9;
    orderBy = 'lastActive';
    UniversityList = [];
    MajorList = [];
    YearRangeList: number[] = [];
    GenderList: string[] = [];

  
    constructor() {
       // this.gender = user.gender === 'female' ? 'male' : 'female'
    }
}

File: API/Helpers/UserParams.cs
using API.Entities;

namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
       
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 100;
        public string OrderBy { get; set; } = "lastActive";
        public List<int> UniversityList { get; set; }
        public List<int> MajorList { get; set; }
        public List<int> YearRangeList { get; set; }
        public List<string> GenderList { get; set; }
    }
}